Test/Install-User.Tests.ps1

```diff
@@ -0,0 +1,293 @@
+            try
+            {
+                $newUser.SID | Should -Be $originalUser.SID
+                $newUser.Description | Should -Be $newDescription
+                $newUser.DisplayName | Should -Be $newFullName
+                $newUser.PasswordNeverExpires | Should -BeFalse
+                $newUser.UserCannotChangePassword | Should -BeTrue
+                Assert-Credential -Password $newPassword
+            }
+            finally
+            {
+                $newUser.Dispose()
+            }
+        }
+        finally
+        {
+            $originalUser.Dispose()
+        }
+    }
+    
+    It 'should update existing users properties with credential' {
+        $fullName = 'Carbon Install User'
+        $credential = New-Credential -Username $username -Password $password
+        $result = Install-CUser -Credential $credential -Description "Original description" -FullName $fullName
+        try
+        {
+            $result | Should -BeNullOrEmpty
+        }
+        finally
+        {
+            if( $result )
+            {
+                $result.Dispose()
+            }
+        }
+    
+        $originalUser = Get-CUser -Username $username
+        $originalUser | Should -Not -BeNullOrEmpty
+        try
+        {
+        
+            $newFullName = 'New {0}' -f $fullName
+            $newDescription = "New description"
+            $newPassword = [Guid]::NewGuid().ToString().Substring(0,14)
+            $credential = New-Credential -UserName $username -Password $newPassword
+            
+            $result = Install-CUser -Credential $credential `
+                                   -Description $newDescription `
+                                   -FullName $newFullName `
+                                   -UserCannotChangePassword `
+                                   -PasswordExpires 
+            try
+            {
+                $result | Should -BeNullOrEmpty
+            }
+            finally
+            {
+                if( $result )
+                {
+                    $result.Dispose()
+                }
+            }
+    
+            [DirectoryServices.AccountManagement.UserPrincipal]$newUser = Get-CUser -Username $username
+            $newUser | Should -Not -BeNullOrEmpty
+            try
+            {
+                $newUser.SID | Should -Be $originalUser.SID
+                $newUser.Description | Should -Be $newDescription
+                $newUser.DisplayName | Should -Be $newFullName
+                $newUser.PasswordNeverExpires | Should -BeFalse
+                $newUser.UserCannotChangePassword | Should -BeTrue
+                Assert-Credential -Password $newPassword
+            }
+            finally
+            {
+                $newUser.Dispose()
+            }
+        }
+        finally
+        {
+            $originalUser.Dispose()
+        }
+    }
+    
+    It 'should allow optional full name' {
+        $fullName = 'Carbon Install User'

```

Comment:
The variable `$fullName` goes unused in this block, shouldn't it be passed to `Install-CUser`?